coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -105,27 +107,66 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
                                    qContext: QueryContext,
                                    forceInProcess: Boolean = false): PlanResult = {
     // Should avoid this asInstanceOf, far many places where we do this now.
-    val params = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
     // MultiPartitionPlanner has capability to stitch across time partitions, however, the logic is mostly broken
     // and not well tested. The logic below would not work well for any kind of subquery since their actual
     // start and ends are different from the start/end parameter of the query context. If we are to implement
     // stitching across time, we need to to pass proper parameters to getPartitions() call
-    val paramToCheckPartitions = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
-    val partitions = getPartitions(logicalPlan, paramToCheckPartitions)
-
-    if (isSinglePartition(partitions)) {
-
-      val (partitionName, startMs, endMs, grpcEndpoint) = partitions.headOption match {
+    if (forceInProcess) {
```

Comment:
IIRC `forceInProcess` is only ever `true` in the `SingleClusterPlanner` when we want to apply a target-schema pushdown. Double-checking: do you set this `true` somewhere? (I don't see any instances in the PR)

Reply:
No, ``forceInProcess`` essentially means in this context is execute all plans including getting raw data from partitions in the query service and not much down to FiloDB.  Also, its set to ``true`` [here](https://github.com/filodb/FiloDB/blob/93c8db6ba4a1e14b12be73f5eebf5b378b691019/coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala#L600)